cli/js/lib.deno.shared_globals.d.ts

```diff
@@ -1046,9 +1046,20 @@ declare class Worker {
     options?: {
       type?: "classic" | "module";
       name?: string;
+      deno?: boolean;
+      permissions?: {
+        read?: boolean;
+        write?: boolean;
+        net?: boolean;
+        env?: boolean;
+        run?: boolean;
+        plugin?: boolean;
+        hrtime?: boolean;
+      };
```

Comment:
I'm thinking that it might be better to nest all Deno specific configs inside `deno` field:
```typescript
options?: {
  type?: "classic" | "module";
  name?: string;
  deno?: {
    permissions?: {
      read?: boolean;
      write?: boolean;
      net?: boolean;
      env?: boolean;
      run?: boolean;
      plugin?: boolean;
      hrtime?: boolean;
    };
  };
}
```
There's plenty other configs we could add like `seed`, `import map`, etc. Waiting on feedback

Reply:
I guess that works... What happens with, for example, Deno.args ? Does it inherit the parent's Deno.args?

Reply:
seed should probably be inherited from the parent if set...

Reply:
> I guess that works... What happens with, for example, Deno.args ? Does it inherit the parent's Deno.args?

Well `Deno.args`, `noColor`, `pid` and other constants are inherited from parent, and I agree that seed should be inherited as well, but there should be a way to specify new seed per worker

Reply:
Actually realized that `crypto` is not supported in workers at all: https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Functions_and_classes_available_to_workers